### PR TITLE
ref(client): Debug, Eq, Hash, PartialEq for ImageLayer; convert annotations to BTreeMap

### PR DESCRIPTION
**Pull request by vdice**
- Update ImageLayer to derive Debug, Eq, Hash and PartialEq traits
- **_Breaking_**: Convert annotations from a HashMap to a BTreeMap per https://github.com/krustlet/oci-distribution/pull/121#discussion_r1544777159, so that we don't need a custom Hash impl for ImageLayer

Changeset motivated by the need to differentiate and deduplicate collections of ImageLayers.

### CODE REVIEW

File: src/client.rs
Comment (thomastaylor312) [inline src/client.rs:136]: I think I'd rather swap annotations to be a `BTreeMap` and break the API a little bit. It makes sense that we'd want these things to be hashable and comparable and the `BTreeMap` is already `Hash` and `Eq`. That way we don't need another dependency and the traits can be entirely derived

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Great change here! One comment around changing the type and just breaking the API

File: src/client.rs
Comment (vdice) [reply src/client.rs]: 👍 Nice, much simpler.  I left the unit test in...

**Review by thomastaylor312 — APPROVED** [review]
Thanks for the great work here! @flavio would like your review too before I merge since this is a breaking API change.

File: src/client.rs
Comment (thomastaylor312) [reply src/client.rs]: Yeah I'm totally fine with itertools as a dev dep

**Comment by thomastaylor312** [comment]
@vdice heads up that it looks like the BTreeMap import didn't get picked up in the windows build for some reason

**Comment by vdice** [comment]
@thomastaylor312 thanks - looks like I didn't update the import in examples/wasm/push.rs.  Now updated; hopefully that was the issue.

**Comment by thomastaylor312** [comment]
Thanks for soldering through that @vdice. Just waiting on a second opinion from @flavio on the breaking aspect of this

**Comment by flavio** [comment]
Sorry, I was on vacation. This change looks good to me

**Comment by vdice** [comment]
Thanks so much @thomastaylor312 and @flavio!

**Pull request merged**
